Client/Frontend/Browser/TabManager.swift

```diff
@@ -265,18 +274,23 @@ extension TabManager {
             if browser.url == nil {
                 return nil
             }
+            if currentItem == nil {
+                return nil
+            }
         }
 
         required init(coder: NSCoder) {
-            self.url = coder.decodeObjectForKey("url") as? NSURL
+            self.urls = coder.decodeObjectForKey("urls") as? [NSURL] ?? [NSURL]()
             self.screenshot = coder.decodeObjectForKey("screenshot") as? UIImage
             self.isSelected = coder.decodeBoolForKey("isSelected")
+            self.currentPage = coder.decodeObjectForKey("currentPage") as? Int ?? 0
         }
 
         func encodeWithCoder(coder: NSCoder) {
-            coder.encodeObject(url, forKey: "url")
+            coder.encodeObject(urls, forKey: "urls")
             coder.encodeObject(screenshot, forKey: "screenshot")
             coder.encodeBool(isSelected, forKey: "isSelected")
+            coder.encodeObject(currentPage, forKey: "currentPage")
```

Comment:
So then you'll want to encode (and decode) the `SessionData` type here instead of separate `urls` and `currentPage` keys.